src/selectize.js

```diff
@@ -1940,7 +1940,11 @@ $.extend(Selectize.prototype, {
 			}
 
 			self.$input.find(old.join(', ')).remove();
-			self.$input.append(fresh.join(''));
+			if (self.caretPos === self.items.length) {
+				self.$input.append(fresh.join(''));
+			} else {
+				$(fresh.join('')).insertBefore(self.$input.children('option')[self.caretPos - 1]);
```

Comment:
What is `fresh.join(''))` and why do we need to construct it here instead of operating on `$input` or `$options`?

Reply:
`fresh` contains composed new <option></option>
`$options` does not exists, `options` contains possible values from the list
please explain what means "operating".
I'm exactly operate with the `$input`.
for some reason I can't make something like this: `self2.$input.insertBefore(new_option, existing_option)`, it destroys the whole `$input`